insights/parsers/krb5.py

```diff
@@ -0,0 +1,110 @@
+     }
+     [libdefaults]
+      dnsdsd = false
+      tilnvs = 24h
+      89EXAsd = {
+       kdc = ks2321serberos.exampsdcsm
+       admin_server = kc5433scderberos.exasdscdm
+     }
+    # renew_lifetime = 7d
+    # forwardable = true
+    # rdns = false
+    '''.strip()
+
+    >>> from insights.tests import context_wrap
+    >>> shared = {Krb5Configuration: Krb5Configuration(context_wrap(krb5_content))}
+    >>> krb5_info = shared[Krb5Configuration]
+    >>> krb5_info.data["libdefaults"]["dnsdsd"]
+    "false"
+    >>> krb5_info.data["realms"]["EXAMPLE.COM"]["kdc"]
+    "kerberos.example.com"
+"""
+
+
+from insights.contrib import pyparsing as p
+from insights.core import Parser
+from insights.core.plugins import parser
+
+
+def parser_squ():
+    lsqu = p.Literal("{").suppress()
+    rsqu = p.Literal("}").suppress()
+    equal = p.Literal("=").suppress()
+    attr_name = subsection_name = p.Word(p.alphas + "_.-/@{}%;:" + p.nums)
+    attr = p.Dict(p.Group(attr_name + equal + p.empty + p.restOfLine))
+    attr_list = p.ZeroOrMore(attr)
+    squ_attr = p.Group(lsqu + attr_list + rsqu)
+    sub_section = p.Dict(p.Group(subsection_name + equal + squ_attr))
+    my_conf = p.Group(p.ZeroOrMore(sub_section))
+    return my_conf
+
+
+@parser('krb5')
+@parser('krb5.conf.d')
+class Krb5Configuration(Parser):
+    """Class for krb5.conf and krb5.conf.d configuration."""
+
+    def parse_content(self, content):
+        dict_all = {}
+        is_squ = False
+        section_name = ""
+        squ_value = []
+        section_value = {}
+        content_mid = []
+        for line in content:
+            if "includedir" in line or line.startswith("#") or not line:
+                continue
+            content_mid.append(line)
+        for line in content_mid:
+            if "[" in line and "]" in line:
+                if not section_name:
+                    section_name = line.strip().strip("[]")
+                    section_value = {}
+                else:
+                    dict_all[section_name] = section_value
+                    section_name = line.strip().strip("[]")
+                    section_value = {}
+            if "=" in line and "{" not in line and "}" not in line and not is_squ:
+                key_value = line.strip().split("=")
+                section_value[key_value[0].strip()] = key_value[1].strip()
+            if "=" in line and "{" in line and "}" in line and not is_squ:
+                key_value = line.strip().split("=")
+                section_value[key_value[0].strip()] = key_value[1].strip()
+            if "=" in line and is_squ:
+                squ_value.append(line)
+            if "=" in line and "{" in line and "}" not in line:
+                is_squ = True
+                squ_value.append(line)
+            if "}" in line and "=" not in line and is_squ:
+                squ_value.append(line)
+                squ_dict = parser_squ().parseString("\n".join(squ_value))[0].asDict()
```

Comment:
I don't think it's a good/efficient method to use the "pyparsing" here. You have already stored the lines in the "{}" pair. And these lines are common "A=B" (if I'm correct), they can be analyzed manually.